adapters/to_sql.py

```diff
@@ -0,0 +1,122 @@
+    state: state enumeration
+    sort_method: sort method enumeration
+    reverse: boolean
+
+  Returns:
+    sql_query: a string which is dynamically constructed using all of the params
+  """
+  sql_query = 'SELECT * FROM dag_run'
+  # convert start and end date from unix to python datetime
+  start_date = datetime.datetime.fromtimestamp(filter_options.start_date)
+  end_date = datetime.datetime.fromtimestamp(filter_options.end_date)
+  # append the date filter to the query based on:
+  # datetype, start_date, and end_date
+  if filter_options.datetype == 0:
+    sql_query += ' WHERE execution_date BETWEEN "' + str(start_date) + '" AND "'+ str(end_date) + '"'  # pylint: disable=line-too-long
+  else:
+    sql_query += ' WHERE execution_date BETWEEN "' + str(start_date) + '" AND "'+ str(end_date) + '"'  # pylint: disable=line-too-long
+  # append a state filter, if there is one available -- '0' means all states
+  if filter_options.state != 0:
+    sql_query += ' AND state = "%s"' %(convert_state(state))
+  # add sorting parameter
+  sql_query = add_sorting(sql_query, filter_options.sort_method, filter_options.reverse)
+  # TODO(dommarques) - add label filtering, probably just the dag_id
+  sql_query += ';'   # put the finishing touch on it
+  return sql_query
+
+
+def to_sql_release(release_id):
+  # get id and execution date from release id
+  dag_id, execution_date = dag_name_parser(release_id)
+  # construct query
+  sql_query = 'SELECT * FROM dag_run'
+  sql_query += ' WHERE dag_id = "' + dag_id + '"'
+  sql_query += ' AND execution_date = "' + str(execution_date) + '"'
+  sql_query += ';'   # put the finishing touch on it
+  return sql_query
+
+
+def to_sql_tasks(execution_date):
+  sql_query = 'SELECT * FROM task_instance'
+  sql_query += ' WHERE execution_date = "' + str(datetime.datetime.fromtimestamp(execution_date)) + '"'
+  sql_query += ' ORDER BY execution_date DESC'
+  sql_query += ';'   # put the finishing touch on it
+  return sql_query
+
+
+def to_sql_task(task_name, execution_date):
+  sql_query = 'SELECT * FROM task_instance'
+  sql_query += ' WHERE execution_date = "' + str(datetime.datetime.fromtimestamp(execution_date)) + '"'
+  sql_query += ' AND task_id = "' + task_name + '"'
+  sql_query += ' ORDER BY execution_date DESC'
+  sql_query += ';'   # put the finishing touch on it
+  return sql_query
+
+
+def to_sql_xcom(dag_id, execution_date):
+  sql_query = 'SELECT * FROM xcom'
+  sql_query += ' WHERE execution_date = "' + str(datetime.datetime.fromtimestamp(execution_date)) + '"'
+  sql_query += ' AND dag_id = "' + dag_id + '"'
+  sql_query += ';'
+  return sql_query
+
+
+def convert_state(state):
+  """Converts the state enumeration into the format needed for the SQL query.
+
+  Args:
+    state: the integer which enumerates the requested state (int)
+
+  Returns:
+    state: string format which follows the same format in the SQl db (str)
+  """
+  state = int(state)
+  if state == 0:
```

Comment:
do not use number, but use the constant representing the number to make it easier to read.